src/Shared/Components/ComponentDescriptorSerializer.Deserialize.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text.Json;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Components
+{
+    // ComponentDescriptorSerializer is an abstraction shared by both MVC and Blazor
+    // This file holds the definition used by Blazor to deserialize descriptors.
+    internal partial class ComponentDescriptorSerializer
+    {
+        public ComponentDescriptorSerializer(
+            IDataProtectionProvider dataProtectionProvider,
+            ILogger<ComponentDescriptorSerializer> logger,
+            RootComponentTypeCache rootComponentTypeCache)
+        {
+            DataProtector = dataProtectionProvider.CreateProtector(DataProtectionProviderPurpose);
+            Logger = logger;
+            RootComponentTypeCache = rootComponentTypeCache;
+        }
+
+        public ILogger<ComponentDescriptorSerializer> Logger { get; }
+
+        public RootComponentTypeCache RootComponentTypeCache { get; }
```

Comment:
Do these really make sense as public properties of this class?

Reply:
Replaced them with private readonly fields. In general I go for property for these things because it doesn't matter (the class is already internal) and the IDE fights me with readonly fields as it adds a `this.` and doesn't prefix the fields with `_` which is annoying because I have to go and manually fix those things.